### ISSUE TITLE
Set log level for all recipes from the config

### ISSUE THREAD

**Issue by joecummings** [pytorch/torchtune#2648]
## Goal

Make it easy to configure the level of logging output you see when training

## Implementation

This should be put in the configs in the section containing metric logging information and profiler information

```yaml
log_level: INFO  # DEBUG, WARN, etc.
```

```python
self._logger = utils.get_logger(cfg.log_level)
```

## Scope

All recipes. 

## Validation

Run a few recipes to confirm that if you use NOTSET, nothing show up, if you use INFO, no debug commands show up, etc.

**Comment by bhargavyagnik** [comment]
I can take this one up.

**Comment by krammnic** [comment]
Let me take this

**Comment by joecummings** [comment]
> Let me take this

My hero
